src/js/view/floating/floating-controller.ts

```diff
@@ -0,0 +1,370 @@
+import type Model from 'controller/model';
+import type Preview from 'view/preview';
+import type { FloatConfig } from 'controller/model';
+import FloatingDragUI from 'view/floating/floating-drag-ui';
+import { OS } from 'environment/environment';
+import { deviceIsLandscape } from 'utils/dom';
+import { isIframe } from 'utils/browser';
+import {
+    addClass,
+    removeClass
+} from 'utils/dom';
+import { STATE_IDLE, STATE_COMPLETE, STATE_ERROR } from 'events/events';
+import { isNumber } from 'utils/underscore';
+import {
+    style,
+} from 'utils/css';
+import viewsManager from 'view/utils/views-manager';
+import type { BoundsObj } from 'types/generic.type';
+import { getPlayerSizeStyles } from 'view/utils/player-size';
+
+const FLOATING_TOP_OFFSET = 62;
+
+let _floatingPlayer: HTMLElement | null = null;
+const fpContainer: {
+    floatingPlayer: HTMLElement | null
+} = {
+    floatingPlayer: _floatingPlayer
+};
+Object.defineProperty(fpContainer, 'floatingPlayer', {
+    get: () => _floatingPlayer,
+    set: (val) => {
+        if (val === _floatingPlayer) {
+            return;
+        }
+        _floatingPlayer = val;
+        watchers.forEach(fc => {
```

Comment:
this covers an edge case where a player is set to `always` float, but upon attempting to `startFloating` it realizes there is another player already floating. if the player stops floating, it should immediately attempt to float once the underlying floating player changes

Reply:
Did you consider forcing the other player to stop floating?

Reply:
If you have two set to always the problem still exists, even without changing it on the fly. Can only have one floating at a time, this preserves the current order of when players are allowed to float (first that tries, gets the prize) and ensures that each player has the chance to float. I made sure to only add this overhead if it is absolutely necessary (see `addWatcher` calls in the code below

Reply:
> If you have two set to always the problem still exists, even without changing it on the fly.

What I am suggesting is that the last one set to always, sets the others to never.

Reply:
Okay so I walked through this with Alicia as a potential solution. We decided to:
1. Stick with the current paradigm by which floating player operates (only one at a time, no override to stop a player from floating, players begin floating as ones before them in the queue stop)
2. Ensure the docs reflect this behavior
3. Provide a code snippet in the docs that allows a developer to code your suggested behavior themselves, it's relatively easy now that we have `getFloating` on the API
4. Solicit feedback after the API goes out to know if we should change that moving forward

Any concerns there? Otherwise this should be good to review functionally as is

Reply:
That's fine. The good thing with this queue, is it will prevent users from tripping themselves up by having an app where too many players all tried to float at once and kept cancelling the other out. The down side is, setFloating('always') will appear to do nothing if another player is floating. If `setFloating` does not actually make a player float immediately, I would want some kind of return value that I could act upon. 

We have a  'float' event that triggers when a player starts and stops floating. If these trigger correctly with the queue then that's fine.